### DISCUSSION TITLE
"Date Submitted" sorting does not match expectations

### DISCUSSION THREAD

**Discussion by marioman63** [ppy/osu#26696] [Ideas] [answered]
When sorting songs by date submitted, the order of the songs is listed in an unexpected order

Here is the very bottom of my beatmap collection, where I expect the oldest songs to be: 
![image](https://github.com/ppy/osu/assets/76608775/d7aa227f-51de-454d-b0a2-cc1060826f85)

the song at the very bottom is https://osu.ppy.sh/beatmapsets/7271#osu/31471 which you can see here has a submission date of 30 May, 2009: 
![image](https://github.com/ppy/osu/assets/76608775/0ac4ab05-2900-4d1c-ad83-384898ac2bde)

However maps older than that are scattered throughout the list. Here is https://osu.ppy.sh/beatmapsets/1#osu/75 (submitted 6 Oct 2007) which is about halfway up the list (note the position of scrollbar): 
![image](https://github.com/ppy/osu/assets/76608775/b83a6aa9-fa5e-4de0-8112-25648578fec6)

And here is the map right below it, submitted 6 November 2020 (https://osu.ppy.sh/beatmapsets/1295033#osu/2969890): 
![image](https://github.com/ppy/osu/assets/76608775/c0f33726-e9d0-4a43-94eb-dae3182a1c56)

I'm not sure what the intended behaviour is beyond sorting them in submission order (as the name implies), i.e., if the top or bottom of the list  should be the oldest, but I'm not exactly sure what it's using as a basis at the moment. I should note that I do have a large number of maps that do not have pages (likely due to being removed), if that is affecting things. There are sections where songs near each other appear sorted, but as shown above, it's not perfect. I can provide more examples if necessary.

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/issues/22416
